frontend/src/composables/useShootSpec.js

```diff
@@ -119,11 +121,20 @@ export function useShootSpec (shootItem, options = {}) {
   })
 
   const isCustomShootDomain = computed(() => {
-    return some(shootDnsProviders.value, 'primary')
+    return !!shootDnsPrimaryProvider.value
   })
 
-  const shootDnsProviders = computed(() => {
-    return get(shootSpec.value, 'dns.providers')
+  const shootDnsPrimaryProvider = computed(() => {
+    return find(shootSpec.value.dns?.providers, { primary: true })
+  })
+  const shootExtensionDnsProviders = computed(() => {
+    const extensionDns = find(shootSpec.value.extensions, { type: 'shoot-dns-service' })
+    return map(extensionDns?.providerConfig.providers, provider => {
```

Comment:
is the `providerConfig` always set?

Reply:
It should be always set. However, if the user manually edits the yaml spec and deletes it.... Same problem as we have in other places, too. We could make it more resilient and check if providerConfig is set, too. WDYT?

Reply:
+1 to make it more resilient